thinkhazard_processing/models.py

```diff
@@ -1,63 +1,117 @@
+    # this value comes from the linked layers
+    calculation_method_quality = Column(Integer)
+    # quality rating for the study
+    # ranges from 0 (bad) to 2 (excellent)
+    # this value comes from the linked layers
+    scientific_quality = Column(Integer)
+
+    # processing steps:
+    # a hazardset starts incomplete.
+    # then it becomes complete, which means:
+    #   * all layers have been downloaded
+    #   * the date, quality, etc fields of the hazardset has been updated
+    complete = Column(Boolean, nullable=False, default=False)
+    # finally it is processed:
+    processed = Column(Boolean, nullable=False, default=False)
 
 
 class Layer(Base):
     __tablename__ = 'layer'
     __table_args__ = {u'schema': 'processing'}
+    # a layer is identified by it's return_period and hazard_set:
+    hazardset_id = Column(String, ForeignKey('processing.hazardset.id'),
+                          primary_key=True)
+    # the return period is typically 100, 475, 2475 years but it can vary
+    return_period = Column(Integer, primary_key=True)
 
-    id = Column(Integer, primary_key=True)
-    # title is the layer title, as reported by geonode, eg: "EQ-PA-100"
-    title = Column(String(1024))
-    hazard_set_id = Column(String(1024),
-                           ForeignKey('processing.dataset.hazard_set_id'))
-    return_period = Column(Integer)  # 100, 475, 2475
-    data_creation_date = Column(Date)
-    md_creation_date = Column(Date)
-    # possible that one of them will not be useful
-    md_update_date = Column(Date)
-    geom = Column(geoalchemy2.Geometry('POLYGON',
-                                       4326,
-                                       management=True))  # bbox ?
-
-    # "downloaded" is set to true by the app
-    # when the layer geotiff has been downloaded
-    # it maintains the application state
-    downloaded = Column(Boolean, default=False)
+    # pixel values have a unit:
+    hazardunit_id = Column(Integer,
+                           ForeignKey('processing.hazardunit.id'),
+                           nullable=False)
+
+    # date the data was last updated (defaults to created):
+    data_lastupdated_date = Column(Date, nullable=False)
+    # date the metadata was last updated (defaults to created):
+    metadata_lastupdated_date = Column(Date, nullable=False)
+
+    # the layer is referenced in geonode with an id:
+    geonode_id = Column(Integer, nullable=False, unique=True)
+    # the data can be downloaded at this URL:
+    download_url = Column(String, nullable=False)
+
+    # quality rating for the hazard calculation method
+    # ranges from 0 (bad) to 10 (excellent)
+    calculation_method_quality = Column(Integer, nullable=False)
+    # quality rating for the study
+    # ranges from 0 (bad) to 2 (excellent)
+    scientific_quality = Column(Integer, nullable=False)
+
+    # "local" is set to false when bounds = -180/-90/180/90
+    # true otherwise
+    local = Column(Boolean, nullable=False)
```

Comment:
When all layers from a given hazardset share the same value of their "local" field, we'll update the hazardset "local" field accordingly. If they do not match, we'll throw an error.